pkg/ruler/ruler.go

```diff
@@ -164,7 +164,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.EnableSharding, "ruler.enable-sharding", false, "Distribute rule evaluation using ring backend")
 	f.StringVar(&cfg.ShardingStrategy, "ruler.sharding-strategy", util.ShardingStrategyDefault, fmt.Sprintf("The sharding strategy to use. Supported values are: %s.", strings.Join(supportedShardingStrategies, ", ")))
 	f.DurationVar(&cfg.FlushCheckPeriod, "ruler.flush-period", 1*time.Minute, "Period with which to attempt to flush rule groups.")
-	f.StringVar(&cfg.RulePath, "ruler.rule-path", "/rules", "file path to store temporary rule files for the prometheus rule managers")
+	f.StringVar(&cfg.RulePath, "ruler.rule-path", "./rules", "file path to store temporary rule files for the prometheus rule managers")
```

Comment:
Not too sure why we use an absolute path there, but if we want to change that, we should mark it as a breaking change

Reply:
Unfortunately we can't change the default config value of a non-experimental config option. What we can do in this PR is rolling back this change and configuring differently `-ruler.rule-path` in `NewRuler()` in E2E tests if required. WDYT?

Reply:
That's the approach I went with